pkg/ring/lifecycler_test.go

```diff
@@ -246,17 +256,90 @@ func TestCheckReady(t *testing.T) {
 	flagext.DefaultValues(&ringConfig)
 	ringConfig.KVStore.Mock = &MockClient{}
 
+	tokens := Tokens([]uint32{1})
 	r, err := New(ringConfig, "ingester")
 	require.NoError(t, err)
 	defer r.Stop()
-	cfg := testLifecyclerConfig(ringConfig, "ring1")
+	cfg := testLifecyclerConfig(t, ringConfig, "ring1")
 	cfg.MinReadyDuration = 1 * time.Nanosecond
 	l1, err := NewLifecycler(cfg, &nopFlushTransferer{}, "ingester")
-	l1.setTokens([]uint32{1})
+	l1.setTokens(tokens)
 	l1.Start()
 	require.NoError(t, err)
 
 	// Delete the ring key before checking ready
 	err = l1.CheckReady(context.Background())
 	require.Error(t, err)
 }
+
+type noopFlushTransferer struct {
+	lifecycler *Lifecycler
+}
+
+func (f *noopFlushTransferer) StopIncomingRequests()                 {}
+func (f *noopFlushTransferer) Flush()                                {}
+func (f *noopFlushTransferer) TransferOut(ctx context.Context) error { return nil }
+
+func TestTokensOnDisk(t *testing.T) {
+	var ringConfig Config
+	flagext.DefaultValues(&ringConfig)
+	ringConfig.KVStore.Mock = consul.NewInMemoryClient(GetCodec())
```

Comment:
You should set `ringConfig.NormaliseTokens = true`, because we're going to get rid of this setting at all (see https://github.com/cortexproject/cortex/pull/1809) and setting it by default.

Reply:
It goes in LifecyclerConfig, so I will add that in `testLifecyclerConfig` instead

Reply:
Apparently `false` is required for `TestRingNormaliseMigration`. Will keep it only for `TestTokensOnDisk`. And good call! `TestTokensOnDisk` fails for `true`.

Reply:
You've marked the comment as resolved, but doesn't look resolved to me.

> Apparently false is required for TestRingNormaliseMigration.

Yes. I think @pstibrany has a PR for it (you should just enable the normalization in your test).